python/tidy/linting_report.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2025 The Servo Project Developers. See the COPYRIGHT
+# file at the top-level directory of this distribution.
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+from dataclasses import dataclass
+from typing import Any, Literal, NotRequired
+
+
+@dataclass
+class GithubAnnotation:
+    file_name: str
+    line_start: int
+    line_end: int
+    level: Literal["notice", "warning", "error"]
+    title: str
+    message: str
+    column_start: NotRequired[int]
+    column_end: NotRequired[int]
+
+
+class GitHubAnnotationManager:
+    def __init__(self, annotation_prefix: str, limit: int = 10):
+        self.annotation_prefix: str = annotation_prefix
+        self.limit: int = limit
+        self.severenty_map: dict[str, Literal["notice", "warning", "error"]] = {
+            "help": "notice",
+            "note": "notice",
+            "warning": "warning",
+            "error": "error",
+        }
```

Comment:
This is only used in `emit_annotations_for_clippy`, so I would move it there and make it a constant. That way this doesn't take up space in every single `GitHubAnnotationManager` that you create.